### DISCUSSION TITLE
Dedicated downloads/metadata/impact for collections

### DISCUSSION THREAD

**Discussion by tefkah** [knowledgefutures/pubpub#1565] [Ideas]
Hi!

I recently worked with a university to publish a book on PubPub. In addition to creating pdfs for the separate chapters/articles (it was an anthology) I also created a full book, which was mostly just a collection of these articles.

One thing I feel that makes PubPub juuust shy of the ideal book publishing platform is the lack of dedicated downloads/"metadata"/impact tracking for collections. I think the "Layouts" view are already a huge improvement over having to put together a page yourself, but I think it would be really cool if we could steal elements from e.g. [OUP Scholarship Online](https://oxford.universitypressscholarship.com/view/10.1093/acprof:oso/9780198746829.001.0001/acprof-9780198746829).

The main three things I'm thinking of are
1. Clear display of the metadata of the book, e.g. ISBN, DOI, authors/editors, title, year, pages, publisher. Of course you can put this together yourself, but a standard way to do this would I think make PubPub a more viable option for publishers.
2. Dedicated download buttons for the entire book (probably just custom downloads, although an auto generated collection of all articles in the collection would be very cool). Going to every page and downloading/reading each article is of course feasible, but having everything in one place is also nice. This can of course be solved by just putting the file there, as I've done (for this book)[https://openpresstiu.pubpub.org/secondthoughts], but that leads to:
3. Impact data for collections. I know that this was not meant to be implemented (#1008), but I think that getting an overview of all the downloads for one collection can be very useful if someone wants to publish multiple collections in the same community. Furthermore, seeing the downloads for the entire PDF/epub is also useful data, but which is not possible to collect through pubpub atm.

Those are my thoughts, thanks for all the work, as always!

**Comment by jeffpooley** [comment]
I just want to second @ThomasFKJorna request for (1) book-related metadata in a standard way, for Book collections; for (2) dedicated download button (but here the issue of multiple-formats (like ePub and Mobi would require some thinking); and (as a result of #2 in part);  (3) true Collection-level metrics, inclusive of full-book downloads.

Right now what we do is throw the metadata at the bottom, and use banners for downloads. We issue each full-book version (PDF, ePub, etc) as a GitHub release, and are able to track downloads through their API. See, for example:

https://www.mediastudies.press/masters-voice

Thanks, @ThomasFKJorna, for articulating this so clearly!

**Comment by gabestein** [comment]
Thanks, both!

1. We should definitely create a collection metadata block for layouts (or modify the collection info block to be more full), and I've [added a card](https://github.com/pubpub/pubpub/issues/1574) for it.

2. We would love to provide both stitched-together downloads for collections and the ability to set a default, same as Pubs. It's on [the roadmap](https://github.com/orgs/pubpub/projects/9#card-52163171), and not a huge left, though at present I don't think we know the timeframe.

3. Thanks for the reminder. We always planned to add an Impact tab for collections, but there were some aggregation issues to solve with identifying what is in a collection and what isn't that we did as part of the collection layouts work. So, I actually think this should be pretty straightforward now, and is also something I can do rather than needing any eng support.

**Reply by johav** [reply]
this will be indeed useful as we (AfricArXiv) need the stats of submissions per repository per collection per country. thanks :)

**Reply by tosteiner** [reply]
👍🏾 +1 ooh, collection-level usage data would be really useful - at COPIM, we have a couple of [reports that rely on PubPub's book collection type](https://copim.pubpub.org/reports), and it'd be amazing to have stats for these.

At the moment, we're relying on Zenodo to provide data on these reports' PDF versions that are provided as separate versions.

**Comment by gabestein** [comment]
Just a heads up that we quietly launched no. 3 above, the Impact dashboard for collections, yesterday. It's still a bit of a work-in-progress...we're still figuring out how to show metrics for the collection "homepage" (whether that's a layout or a page), and there are some issues with the default sorting I'm working through. But the Pub info is up and should be accurate. Let me know what you think!

**Reply by jeffpooley** [reply]
This is fabulous—so great to see this implemented, so quickly. The layout looks great!

**Comment by gabestein** [comment]
No. 1 above, a better metadata display option for collections, is now available as well in the collection header block on collection layouts.

<img width="1077" alt="Screen Shot 2021-10-27 at 13 47 24" src="https://user-images.githubusercontent.com/639110/139118841-72a7ffdf-5172-4592-9c3b-29d6ef09d966.png">

**Reply by tefkah** [reply]
Fantastic!! As always, beyond thrilled by your responsiveness, thank you so much!

**Comment by tefkah** [comment]
Hi again, I had some further suggestions for this page if you would be open for that! 
There's probably technical limitations involved, as most of my questions would involve making the Layout more like a Pub, and from my glances at the codebase these seem to be treated quite differently, but nevertheless!

1. It would be really nice to be able to attach a custom pdf to this "page" and track the amount of downloads it generates. Atm there's no good place to place a full book so we resort (see https://doi.org/10.26116/1h0x-9k96) to just slapping the pdf there, but you lose the ability to track the downloads. Of course we could set this up somewhere differently ourselves, but that kind of defeats the point of using PubPub as the main publishing platform.
2. Currently almost no metadata about the collection is available through the `<meta>` tags, theyre just the same as the ones for a page. Ideally for issues and books at least, the meta tags would get populated in a similar way to how Pubs are. For Tag collections this is probably not necessary.
3. The DOI field gets duplicated and does not indicate that it's a doi. 
![image](https://user-images.githubusercontent.com/21983833/176894657-adb13010-aa7b-4873-975e-60120ed863fe.png)

If you add the https://doi.org yourself ([as is required by Crossref's DOI display guidelines](https://www.crossref.org/display-guidelines/), the link no longer works.
![image](https://user-images.githubusercontent.com/21983833/176894358-bc13b584-80d3-4cee-89f0-36c31a341d63.png)


One possible solution for this might be to make the layout page a certain kind of book/issue Pub which also allows you to display its entries instead. Again, they work very differently so this may be infeasible.

Thanks for listening to another rant!

**Reply by tefkah** [reply]
For those interested, I solved the latter issue with the following two CSS snippets
```css
.details-row > .click-to-copy + div {
    display: none;
}

.details-row .click-to-copy a::before { 
    content: "https://doi.org/"
}
```
